src/js/player.js

```diff
@@ -2062,6 +2070,26 @@ class Player extends Component {
   }
 
   /**
+   * Returns the current source objects.
+   *
+   * @return {Object[]} The current source objects
+   * @method currentSources
+   */
+  currentSources() {
+    return this.cache_.sources || [this.currentSource()];
+  }
+
+  /**
+   * Returns the current source object.
+   *
+   * @return {Object} The current source object
+   * @method currentSource
+   */
+  currentSource() {
+    return this.cache_.source || { src: this.currentSrc() };
```

Comment:
Unfortunately, `undefined` would be not backwards compatible but also would break away from what the video element does for `currentSrc`:
```js
document.createElement('video').currentSrc
// ""
```

As for the others, returning an empty object and empty array would make it consistent. Though, would be a bit annoying to check for whether it's empty or not.